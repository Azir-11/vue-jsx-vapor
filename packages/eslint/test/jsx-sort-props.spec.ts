import { createRuleTester } from 'eslint-vitest-rule-tester'
import { describe, expect, it } from 'vitest'
import jsxSortProps from '../src/rules/jsx-sort-props'

describe('jsx-sort-props', () => {
  const { invalid } = createRuleTester({
    name: 'jsx-sort-props',
    rule: jsxSortProps,
    parserOptions: {
      sourceType: 'module',
      ecmaFeatures: {
        jsx: true,
      },
    },
  })

  it('basic', async () => {
    const { result } = await invalid({
      code: `
        <div b a />
      `,
      errors: ['sortPropsByAlpha'],
    })
    expect(result.output).toMatchSnapshot()
  })

  it('reservedFirst', async () => {
    const { result } = await invalid({
      code: '<App a v-for={i in 4} v-if={true} b />',
      errors: ['listReservedPropsFirst', 'listReservedPropsFirst'],
      options: [
        { reservedFirst: ['v-if', 'v-for'], noSortAlphabetically: true },
      ],
    })
    expect(result.output).toMatchSnapshot()
  })

  it('reservedLast', async () => {
    const { result } = await invalid({
      code: '<App v-slot={{ foo }} onClick={() => {}} />',
      errors: ['listReservedPropsLast'],
      options: [{ reservedLast: ['v-slot'], callbacksLast: true }],
    })
    expect(result.output).toMatchSnapshot()
  })

  it('reservedFirst and reservedLast', async () => {
    const { result } = await invalid({
      code: '<App a v-model:b={foo} b v-model={foo} c v-slot:b={{ foo }} v-slot:a={{ foo }} onClick={() => {}} />',
      errors: [
        'listReservedPropsFirst',
        'listReservedPropsFirst',
        'sortPropsByAlpha',
        'listReservedPropsLast',
      ],
      options: [{ reservedFirst: ['v-model'], reservedLast: ['v-slot'] }],
    })
    expect(result.output).toMatchSnapshot()
  })
})
