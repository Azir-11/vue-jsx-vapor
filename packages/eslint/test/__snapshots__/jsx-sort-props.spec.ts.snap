// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`jsx-sort-props > basic 1`] = `
"
        <div a b />
      "
`;

exports[`jsx-sort-props > reservedFirst 1`] = `"<App v-if={true} v-for={i in 4} a b />"`;

exports[`jsx-sort-props > reservedFirst and reservedLast 1`] = `"<App v-model={foo} v-model:b={foo} a b c onClick={() => {}} v-slot:a={{ foo }} v-slot:b={{ foo }} />"`;

exports[`jsx-sort-props > reservedLast 1`] = `"<App onClick={() => {}} v-slot={{ foo }} />"`;
